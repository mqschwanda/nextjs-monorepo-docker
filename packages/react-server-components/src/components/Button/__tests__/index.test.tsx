import { render } from '@testing-library/react';
import { Button } from '..';

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Button />', () => {
      it('renders', () => {
        const testId = 'Button';
        const chilren = 'children...';

        const { getByTestId, asFragment } = render((
          <Button
            className='test'
            testId={testId}
          >
            { chilren }
          </Button>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
