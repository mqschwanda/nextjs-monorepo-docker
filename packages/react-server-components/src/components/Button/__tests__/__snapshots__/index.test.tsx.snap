// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Button /> renders 1`] = `
<DocumentFragment>
  <button
    class="btn btn-md test"
    data-testid="Button"
    type="button"
  >
    children...
  </button>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Color Example 1`] = `
[
  <button
    className="btn btn-accent btn-md"
    type="button"
  >
    accent
  </button>,
  <button
    className="btn btn-error btn-md"
    type="button"
  >
    error
  </button>,
  <button
    className="btn btn-ghost btn-md"
    type="button"
  >
    ghost
  </button>,
  <button
    className="btn glass btn-md"
    type="button"
  >
    glass
  </button>,
  <button
    className="btn btn-info btn-md"
    type="button"
  >
    info
  </button>,
  <button
    className="btn btn-link btn-md"
    type="button"
  >
    link
  </button>,
  <button
    className="btn btn-neutral btn-md"
    type="button"
  >
    neutral
  </button>,
  <button
    className="btn btn-primary btn-md"
    type="button"
  >
    primary
  </button>,
  <button
    className="btn btn-secondary btn-md"
    type="button"
  >
    secondary
  </button>,
  <button
    className="btn btn-success btn-md"
    type="button"
  >
    success
  </button>,
  <button
    className="btn btn-warning btn-md"
    type="button"
  >
    warning
  </button>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Example 1`] = `
<button
  className="btn btn-md"
  type="button"
>
  Click Me!
</button>
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Loading Example 1`] = `
<button
  className="btn loading btn-md"
  type="button"
>
  Loading
</button>
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button No Animation Example 1`] = `
<button
  className="btn no-animation btn-md"
  type="button"
>
  No Animation
</button>
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Shape Example 1`] = `
[
  <button
    className="btn btn-circle btn-md"
    type="button"
  >
    c
  </button>,
  <button
    className="btn btn-square btn-md"
    type="button"
  >
    s
  </button>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Size Example 1`] = `
[
  <button
    className="btn btn-lg"
    type="button"
  >
    lg
  </button>,
  <button
    className="btn btn-md"
    type="button"
  >
    md
  </button>,
  <button
    className="btn btn-sm"
    type="button"
  >
    sm
  </button>,
  <button
    className="btn btn-xs"
    type="button"
  >
    xs
  </button>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/Button Button Wide Example 1`] = `
<button
  className="btn btn-wide btn-md"
  type="button"
>
  Wide
</button>
`;
